### PR TITLE
Added Multispan<T>

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
Very useful when dealing with I/O traffic coming in batches.

### CODE REVIEW

**Comment by KrzysztofCwalina** [comment]
@sokket

File: src/System.Buffers.Experimental/System/Buffers/Multispan.cs [hunk 1]
```diff
@@ -0,0 +1,213 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Buffers
+{
+    /// <summary>
+    /// A collection of pooled arrays.
```
Comment (sokket) [inline src/System.Buffers.Experimental/System/Buffers/Multispan.cs]: This is confusing...what scenario does this allow that the regular pool does not? It looks like (and this comment specifies) that this is just a collection of different size, pooled ```T```s.

**Comment by omariom** [comment]
How is it intended  to be used?

**Comment by KrzysztofCwalina** [comment]
@omariom, see the doc comment at the type level.
It might be not the greatest doc, but I think it will point you in the right direction. 

The main usage is in async i/o. Calling code allocates new segment before each i/o call (ReceiveAsync, for example); the i/o operation fills in the next segment with data that was received in current chunk, and then you repeat till i/o is completed. 

The same cannot be achieved with a collection of spans, because spans should not be on the heap. It also cannot be achieved with a collection of arrays, because user code could store the pooled arrays for later (and what's worse return them to the pool multiple times). 

Soon I should be able to check in sample showing how it all comes together in an iocp service.

Maybe Multispan is not the greatest name, but we had a type like that in an incubation project I worked on, and so I reused it.

**Comment by omariom** [comment]
What if one wants to use unmanaged chunks of memory for IO? 
 ArrayPool allocates from managed heap. Can it be abstracted?

**Comment by KrzysztofCwalina** [comment]
Yeah, we should think about ArrayPool like abstraction that gives off Span<T>s instead of T[]s. Once we have such abstraction, we should add Multispan ctor that takes such abstract pool and uses it instead of using ArrayPool<T>.Shared.

**Pull request merged**